src/vhip/mlmodel/gene_features.py

```diff
@@ -160,13 +160,58 @@ def __init__(self, gene_file: str) -> None:
             raise Exception(
                 "Genes file is not provided or empty. Please provide a valid gene file."
             )
-
+        self.id = gene_file
+        self.genes: List[Gene] = []
+        self.skipped_genes: List[str] = []
         readout = read_annotated_genes(gene_file)
-        self.genes: List[Gene] = [
-            Gene(
-                gene_seq=str(readout[0][out]),
-                gene_id=str(readout[1][out]),
-                gene_product=str(readout[2][out]),
+
+        for out in range(len(readout[0])):
+            try:
+                self.genes.append(
+                    Gene(
+                        gene_seq=str(readout[0][out]),
+                        gene_id=str(readout[1][out]),
+                        gene_product=str(readout[2][out]),
+                    )
+                )
+            except Exception:
+                self.skipped_genes.append(str(readout[1][out]))
+        print(f"{len(self.skipped_genes)} genes skipped")
+
+    def codon_counts(
+        self, threshold_imprecise: int, threshold_skipped_genes: int
+    ) -> None:
```

Comment:
set a default for the threshold!

Reply:
for both threshold

Reply:
Ok. I could either use a really high number or try to figure out what no limit (infinity) would look like in code

Reply:
Why would you recommend a high value? Do you know the distribution of imprecise codons? I'd start there and decide on a threshold to use from there!

Reply:
I do not know the distribution. I can go on great lakes and rerun some code to get just that info. Do you think I should choose something like top quartile?

Reply:
I'd choose a number that would work for most gene files, but might fail for files that have poor genes prediction all around!I don't know what that number is, so looking at the distribution of imprecise codons should help pick a number!

Reply:
chose 0 for threshold_imprecise, and 0.5 for threshold_skipped genes. Reasoning discussed over slack